{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "path = \"/home/ubuntu/honors_research/depmap/K562_mutations.csv\"\n",
    "mutations = pd.read_csv(path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>Variant Type</th>\n",
       "      <th>Variant Info</th>\n",
       "      <th>Ref Allele</th>\n",
       "      <th>Alt Allele</th>\n",
       "      <th>Allele Fraction</th>\n",
       "      <th>Ref Count</th>\n",
       "      <th>Alt Count</th>\n",
       "      <th>...</th>\n",
       "      <th>Vep Somatic</th>\n",
       "      <th>LoF Number of Transcripts in Gene</th>\n",
       "      <th>Vep Impact</th>\n",
       "      <th>Oncogene High Impact</th>\n",
       "      <th>AM class</th>\n",
       "      <th>AM Pathogenicity</th>\n",
       "      <th>Hotspot</th>\n",
       "      <th>Intron</th>\n",
       "      <th>Exon</th>\n",
       "      <th>Rescue Reason</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CHD5</td>\n",
       "      <td>chr1</td>\n",
       "      <td>6121138</td>\n",
       "      <td>substitution</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>TC</td>\n",
       "      <td>GA</td>\n",
       "      <td>0.426</td>\n",
       "      <td>31</td>\n",
       "      <td>23</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33/42</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CHD5</td>\n",
       "      <td>chr1</td>\n",
       "      <td>6136538</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.242</td>\n",
       "      <td>48</td>\n",
       "      <td>15</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>False</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.9967</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17/42</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TNFRSF9</td>\n",
       "      <td>chr1</td>\n",
       "      <td>7939993</td>\n",
       "      <td>SNV</td>\n",
       "      <td>start_lost</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.729</td>\n",
       "      <td>13</td>\n",
       "      <td>38</td>\n",
       "      <td>...</td>\n",
       "      <td>0&amp;1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8-Feb</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CASZ1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10653866</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.488</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>False</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.1163</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21-Nov</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PLA2G2E</td>\n",
       "      <td>chr1</td>\n",
       "      <td>19920311</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.246</td>\n",
       "      <td>32</td>\n",
       "      <td>11</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>False</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.9222</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4-Apr</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>293</th>\n",
       "      <td>ZXDA</td>\n",
       "      <td>chrX</td>\n",
       "      <td>57909058</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.400</td>\n",
       "      <td>11</td>\n",
       "      <td>8</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>False</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>0.4213</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1-Jan</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294</th>\n",
       "      <td>ZXDA</td>\n",
       "      <td>chrX</td>\n",
       "      <td>57909607</td>\n",
       "      <td>deletion</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>CCTCGGGGCACAGGT</td>\n",
       "      <td>C</td>\n",
       "      <td>0.417</td>\n",
       "      <td>13</td>\n",
       "      <td>10</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1-Jan</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295</th>\n",
       "      <td>MTMR8</td>\n",
       "      <td>chrX</td>\n",
       "      <td>64268590</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.935</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>False</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.0655</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>14/14</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296</th>\n",
       "      <td>MTMR8</td>\n",
       "      <td>chrX</td>\n",
       "      <td>64336111</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.214</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>False</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.3345</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>14-Oct</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297</th>\n",
       "      <td>TEX13C</td>\n",
       "      <td>chrX</td>\n",
       "      <td>125322182</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.250</td>\n",
       "      <td>21</td>\n",
       "      <td>7</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>False</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.7715</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2-Jan</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>298 rows × 68 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene Chromosome   Position  Variant Type        Variant Info  \\\n",
       "0       CHD5       chr1    6121138  substitution    missense_variant   \n",
       "1       CHD5       chr1    6136538           SNV    missense_variant   \n",
       "2    TNFRSF9       chr1    7939993           SNV          start_lost   \n",
       "3      CASZ1       chr1   10653866           SNV    missense_variant   \n",
       "4    PLA2G2E       chr1   19920311           SNV    missense_variant   \n",
       "..       ...        ...        ...           ...                 ...   \n",
       "293     ZXDA       chrX   57909058           SNV    missense_variant   \n",
       "294     ZXDA       chrX   57909607      deletion  frameshift_variant   \n",
       "295    MTMR8       chrX   64268590           SNV    missense_variant   \n",
       "296    MTMR8       chrX   64336111           SNV    missense_variant   \n",
       "297   TEX13C       chrX  125322182           SNV    missense_variant   \n",
       "\n",
       "          Ref Allele Alt Allele  Allele Fraction  Ref Count  Alt Count  ...  \\\n",
       "0                 TC         GA            0.426         31         23  ...   \n",
       "1                  A          G            0.242         48         15  ...   \n",
       "2                  A          G            0.729         13         38  ...   \n",
       "3                  G          T            0.488         25         25  ...   \n",
       "4                  C          T            0.246         32         11  ...   \n",
       "..               ...        ...              ...        ...        ...  ...   \n",
       "293                C          A            0.400         11          8  ...   \n",
       "294  CCTCGGGGCACAGGT          C            0.417         13         10  ...   \n",
       "295                T          C            0.935          0         14  ...   \n",
       "296                T          G            0.214         10          2  ...   \n",
       "297                G          T            0.250         21          7  ...   \n",
       "\n",
       "    Vep Somatic  LoF Number of Transcripts in Gene Vep Impact  \\\n",
       "0           NaN                                NaN   MODERATE   \n",
       "1             1                                NaN   MODERATE   \n",
       "2           0&1                                1.0       HIGH   \n",
       "3             1                                NaN   MODERATE   \n",
       "4           NaN                                NaN   MODERATE   \n",
       "..          ...                                ...        ...   \n",
       "293         NaN                                NaN   MODERATE   \n",
       "294         NaN                                1.0       HIGH   \n",
       "295         NaN                                NaN   MODERATE   \n",
       "296         NaN                                NaN   MODERATE   \n",
       "297         NaN                                NaN   MODERATE   \n",
       "\n",
       "    Oncogene High Impact           AM class AM Pathogenicity Hotspot  Intron  \\\n",
       "0                  False                NaN              NaN   False     NaN   \n",
       "1                  False  likely_pathogenic           0.9967   False     NaN   \n",
       "2                  False                NaN              NaN   False     NaN   \n",
       "3                  False      likely_benign           0.1163   False     NaN   \n",
       "4                  False  likely_pathogenic           0.9222   False     NaN   \n",
       "..                   ...                ...              ...     ...     ...   \n",
       "293                False          ambiguous           0.4213   False     NaN   \n",
       "294                False                NaN              NaN   False     NaN   \n",
       "295                False      likely_benign           0.0655   False     NaN   \n",
       "296                False      likely_benign           0.3345   False     NaN   \n",
       "297                False  likely_pathogenic           0.7715   False     NaN   \n",
       "\n",
       "       Exon  Rescue Reason  \n",
       "0     33/42            NaN  \n",
       "1     17/42            NaN  \n",
       "2     8-Feb            NaN  \n",
       "3    21-Nov            NaN  \n",
       "4     4-Apr            NaN  \n",
       "..      ...            ...  \n",
       "293   1-Jan            NaN  \n",
       "294   1-Jan            NaN  \n",
       "295   14/14            NaN  \n",
       "296  14-Oct            NaN  \n",
       "297   2-Jan            NaN  \n",
       "\n",
       "[298 rows x 68 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#display third column\n",
    "mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>Variant Type</th>\n",
       "      <th>Variant Info</th>\n",
       "      <th>Ref Allele</th>\n",
       "      <th>Alt Allele</th>\n",
       "      <th>Allele Fraction</th>\n",
       "      <th>Ref Count</th>\n",
       "      <th>Alt Count</th>\n",
       "      <th>...</th>\n",
       "      <th>Vep Somatic</th>\n",
       "      <th>LoF Number of Transcripts in Gene</th>\n",
       "      <th>Vep Impact</th>\n",
       "      <th>Oncogene High Impact</th>\n",
       "      <th>AM class</th>\n",
       "      <th>AM Pathogenicity</th>\n",
       "      <th>Hotspot</th>\n",
       "      <th>Intron</th>\n",
       "      <th>Exon</th>\n",
       "      <th>Rescue Reason</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CHD5</td>\n",
       "      <td>chr1</td>\n",
       "      <td>6121138</td>\n",
       "      <td>substitution</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>TC</td>\n",
       "      <td>GA</td>\n",
       "      <td>0.426</td>\n",
       "      <td>31</td>\n",
       "      <td>23</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33/42</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CHD5</td>\n",
       "      <td>chr1</td>\n",
       "      <td>6136538</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.242</td>\n",
       "      <td>48</td>\n",
       "      <td>15</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>False</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.9967</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17/42</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TNFRSF9</td>\n",
       "      <td>chr1</td>\n",
       "      <td>7939993</td>\n",
       "      <td>SNV</td>\n",
       "      <td>start_lost</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.729</td>\n",
       "      <td>13</td>\n",
       "      <td>38</td>\n",
       "      <td>...</td>\n",
       "      <td>0&amp;1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8-Feb</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CASZ1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10653866</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.488</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>False</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.1163</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21-Nov</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PLA2G2E</td>\n",
       "      <td>chr1</td>\n",
       "      <td>19920311</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.246</td>\n",
       "      <td>32</td>\n",
       "      <td>11</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>False</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.9222</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4-Apr</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>292</th>\n",
       "      <td>TSR2</td>\n",
       "      <td>chrX</td>\n",
       "      <td>54444542</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.954</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>...</td>\n",
       "      <td>0&amp;1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>False</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.8153</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5-May</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>293</th>\n",
       "      <td>ZXDA</td>\n",
       "      <td>chrX</td>\n",
       "      <td>57909058</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.400</td>\n",
       "      <td>11</td>\n",
       "      <td>8</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>False</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>0.4213</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1-Jan</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295</th>\n",
       "      <td>MTMR8</td>\n",
       "      <td>chrX</td>\n",
       "      <td>64268590</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.935</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>False</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.0655</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>14/14</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296</th>\n",
       "      <td>MTMR8</td>\n",
       "      <td>chrX</td>\n",
       "      <td>64336111</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.214</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>False</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.3345</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>14-Oct</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297</th>\n",
       "      <td>TEX13C</td>\n",
       "      <td>chrX</td>\n",
       "      <td>125322182</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.250</td>\n",
       "      <td>21</td>\n",
       "      <td>7</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>False</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.7715</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2-Jan</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>284 rows × 68 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene Chromosome   Position  Variant Type      Variant Info Ref Allele  \\\n",
       "0       CHD5       chr1    6121138  substitution  missense_variant         TC   \n",
       "1       CHD5       chr1    6136538           SNV  missense_variant          A   \n",
       "2    TNFRSF9       chr1    7939993           SNV        start_lost          A   \n",
       "3      CASZ1       chr1   10653866           SNV  missense_variant          G   \n",
       "4    PLA2G2E       chr1   19920311           SNV  missense_variant          C   \n",
       "..       ...        ...        ...           ...               ...        ...   \n",
       "292     TSR2       chrX   54444542           SNV  missense_variant          A   \n",
       "293     ZXDA       chrX   57909058           SNV  missense_variant          C   \n",
       "295    MTMR8       chrX   64268590           SNV  missense_variant          T   \n",
       "296    MTMR8       chrX   64336111           SNV  missense_variant          T   \n",
       "297   TEX13C       chrX  125322182           SNV  missense_variant          G   \n",
       "\n",
       "    Alt Allele  Allele Fraction  Ref Count  Alt Count  ... Vep Somatic  \\\n",
       "0           GA            0.426         31         23  ...         NaN   \n",
       "1            G            0.242         48         15  ...           1   \n",
       "2            G            0.729         13         38  ...         0&1   \n",
       "3            T            0.488         25         25  ...           1   \n",
       "4            T            0.246         32         11  ...         NaN   \n",
       "..         ...              ...        ...        ...  ...         ...   \n",
       "292          G            0.954          0         19  ...         0&1   \n",
       "293          A            0.400         11          8  ...         NaN   \n",
       "295          C            0.935          0         14  ...         NaN   \n",
       "296          G            0.214         10          2  ...         NaN   \n",
       "297          T            0.250         21          7  ...         NaN   \n",
       "\n",
       "     LoF Number of Transcripts in Gene Vep Impact Oncogene High Impact  \\\n",
       "0                                  NaN   MODERATE                False   \n",
       "1                                  NaN   MODERATE                False   \n",
       "2                                  1.0       HIGH                False   \n",
       "3                                  NaN   MODERATE                False   \n",
       "4                                  NaN   MODERATE                False   \n",
       "..                                 ...        ...                  ...   \n",
       "292                                NaN   MODERATE                False   \n",
       "293                                NaN   MODERATE                False   \n",
       "295                                NaN   MODERATE                False   \n",
       "296                                NaN   MODERATE                False   \n",
       "297                                NaN   MODERATE                False   \n",
       "\n",
       "              AM class AM Pathogenicity Hotspot  Intron    Exon  Rescue Reason  \n",
       "0                  NaN              NaN   False     NaN   33/42            NaN  \n",
       "1    likely_pathogenic           0.9967   False     NaN   17/42            NaN  \n",
       "2                  NaN              NaN   False     NaN   8-Feb            NaN  \n",
       "3        likely_benign           0.1163   False     NaN  21-Nov            NaN  \n",
       "4    likely_pathogenic           0.9222   False     NaN   4-Apr            NaN  \n",
       "..                 ...              ...     ...     ...     ...            ...  \n",
       "292  likely_pathogenic           0.8153   False     NaN   5-May            NaN  \n",
       "293          ambiguous           0.4213   False     NaN   1-Jan            NaN  \n",
       "295      likely_benign           0.0655   False     NaN   14/14            NaN  \n",
       "296      likely_benign           0.3345   False     NaN  14-Oct            NaN  \n",
       "297  likely_pathogenic           0.7715   False     NaN   2-Jan            NaN  \n",
       "\n",
       "[284 rows x 68 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# filtering for snps (SNV or substiutition)\n",
    "mutations = mutations[mutations[\"Variant Type\"].isin([\"SNV\", \"substitution\"])]\n",
    "mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_803739/2454380929.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mutations_new[\"start\"] = mutations[\"Position\"] - 1\n",
      "/tmp/ipykernel_803739/2454380929.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mutations_new[\"end\"] = mutations[\"Position\"]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6121137</td>\n",
       "      <td>6121138</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6136537</td>\n",
       "      <td>6136538</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7939992</td>\n",
       "      <td>7939993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10653865</td>\n",
       "      <td>10653866</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>19920310</td>\n",
       "      <td>19920311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>292</th>\n",
       "      <td>chrX</td>\n",
       "      <td>54444541</td>\n",
       "      <td>54444542</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>293</th>\n",
       "      <td>chrX</td>\n",
       "      <td>57909057</td>\n",
       "      <td>57909058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295</th>\n",
       "      <td>chrX</td>\n",
       "      <td>64268589</td>\n",
       "      <td>64268590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296</th>\n",
       "      <td>chrX</td>\n",
       "      <td>64336110</td>\n",
       "      <td>64336111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297</th>\n",
       "      <td>chrX</td>\n",
       "      <td>125322181</td>\n",
       "      <td>125322182</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>284 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Chromosome      start        end\n",
       "0         chr1    6121137    6121138\n",
       "1         chr1    6136537    6136538\n",
       "2         chr1    7939992    7939993\n",
       "3         chr1   10653865   10653866\n",
       "4         chr1   19920310   19920311\n",
       "..         ...        ...        ...\n",
       "292       chrX   54444541   54444542\n",
       "293       chrX   57909057   57909058\n",
       "295       chrX   64268589   64268590\n",
       "296       chrX   64336110   64336111\n",
       "297       chrX  125322181  125322182\n",
       "\n",
       "[284 rows x 3 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# taking chromosome + position columns + making a new column (for end position)\n",
    "mutations_new = mutations[[\"Chromosome\"]]\n",
    "mutations_new[\"start\"] = mutations[\"Position\"] - 1\n",
    "mutations_new[\"end\"] = mutations[\"Position\"]\n",
    "mutations_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "# export to bed\n",
    "mutations_new.to_csv(\"K562_mutations.bed\", sep=\"\\t\", header=False, index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
